feat: 4947 - added "download language" to product table

monsieurtanuki opened the pull request:
### What
- Now, each time we download one or several products, we store the download language too.
- "This product was downloaded using an API query with Italian as the language".
- With this additional metadata, we'll be able to refresh the local database on specific products for localization reasons like "only products downloaded in a different language".
  - either as a silent background task ("just refresh all the products that are not in that language")
  - or as user hint when landing on the product page ("hey, this product was downloaded in Italian, would you like to refresh it in English?")
  - or as user hint when landing on the product list page ("hey, several products on this page were downloaded in Italian, would you like to refresh the page in English?")
  - or a combination of all this

### Part of 
- #4947

### Files
Deleted file:
* `dao_product_migration.dart`

Impacted files:
* `background_task_download_products.dart`: added language parameter
* `dao_hive_product.dart`: removed dead code
* `dao_product.dart`: added table column language `lc`
* `local_database.dart`: upgraded the database version
* `onboarding_data_product.dart`: added language parameter
* `product_list_page.dart`: added language parameter
* `product_refresher.dart`: added language parameter
* `query_product_list_supplier.dart`: added language parameter

monsieurtanuki committed:
feat: 4947 - added "download language" to product table

Deleted file:
* `dao_product_migration.dart`

Impacted files:
* `background_task_download_products.dart`: added language parameter
* `dao_hive_product.dart`: removed dead code
* `dao_product.dart`: added table column language `lc`
* `local_database.dart`: upgraded the database version
* `onboarding_data_product.dart`: added language parameter
* `product_list_page.dart`: added language parameter
* `product_refresher.dart`: added language parameter
* `query_product_list_supplier.dart`: added language parameter

M123-dev commented on packages/smooth_app/lib/database/dao_hive_product.dart:
```diff
@@ -35,48 +34,4 @@ class DaoHiveProduct extends AbstractDao implements DaoProductMigrationSource {
-
-  @override
-  Future<Map<String, Product>> getAll(final List<String> barcodes) async {
-    final LazyBox<Product> box = _getBox();
-    final Map<String, Product> result = <String, Product>{};
-    for (final String barcode in barcodes) {
-      final Product? product = await box.get(barcode);
-      if (product != null) {
-        result[barcode] = product;
-      }
-    }
-    return result;
-  }
-
-  Future<void> put(final Product product) async => putAll(<Product>[product]);
-
-  Future<void> putAll(final Iterable<Product> products) async {
-    final Map<String, Product> upserts = <String, Product>{};
-    for (final Product product in products) {
-      upserts[product.barcode!] = product;
-    }
-    await _getBox().putAll(upserts);
-  }
-
-  @override
-  Future<List<String>> getAllKeys() async {
-    final LazyBox<Product> box = _getBox();
-    final List<String> result = <String>[];
-    for (final dynamic key in box.keys) {
-      result.add(key.toString());
-    }
-    return result;
-  }
-
-  // Just for the migration
-  @override
-  Future<void> deleteAll(final List<String> barcodes) async {
-    final LazyBox<Product> box = _getBox();
-    await box.deleteAll(barcodes);
-  }
```
Did you test if this part removal will work or also break the whole app, nontheless we should keep an eye out in the beta

M123-dev reviewed (approved):
Thanks a lot @monsieurtanuki

monsieurtanuki commented:
Thank you @M123-dev for your review!
I understand your concern about dead code removal when it comes to `hive`. I've removed code that is not "`hive` native" code, I've managed to upgrade an existing app and to run an app from scratch. Therefore I'm rather confident.
But you're right, we should be cautious.
For the record, it was not (only) for revenge that I removed that code, but because the notion of "product download language" didn't exist back then.
